### PR TITLE
[Docker Desktop Extension] Make Catalog Cards Clickable

### LINKED ISSUE #11517: [Docker Desktop Extension] Make the Catalogue Display Items Clickable
- Make these designs clickable,opening the single page catalog item. Create a separate button to open the entire Catalog. 
- If Compose Apps are pre-imported, then the Catalog area can show their imported designs.

https://docs.google.com/document/d/1sxO8rWemfzDFUEh8QCBKx-4uKlXy7KBYOekK_fdkLGI/edit

### PR DESCRIPTION
**Pull request by humblenginr**
**Notes for Reviewers**

This PR fixes #11517 

This PR also removes some redundancies in fetching catalog entities and refactors the catalog chart logic to dynamically use the data from the backend.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by humblenginr** [comment]
https://github.com/user-attachments/assets/537616ed-6a29-409f-b66f-31886ed89ccb

File: install/docker-extension/ui/src/components/Catalog/CatalogCard.js [hunk 1]
```diff
@@ -25,9 +25,10 @@ function CatalogCard({ pattern, patternType, catalog }) {
   const handleCopyClick = async event => {
     event.preventDefault();
 
+    let name = pattern?.name?.toLowerCase().split(' ').join('-')
     try {
       await navigator.clipboard.writeText(
-        `${mesheryCloudUrl}/catalog/${pattern.id}`
+        `${mesheryCloudUrl}/catalog/content/${name}-${pattern.id}?source=%257B%257D`
```
Comment (leecalcote) [inline install/docker-extension/ui/src/components/Catalog/CatalogCard.js:31]: When this name changes, the link will break.

@Yashsharma1911, @captain-Akshay 


@humblenginr use this format instead - http://meshery.layer5.io/catalog/content/design/3a0dfed8-e5ff-4d47-94c4-af4e22c9704a

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: install/docker-extension/ui/package.json
Comment (leecalcote) [inline install/docker-extension/ui/package.json:1]: This change should not have been committed.

File: install/docker-extension/ui/src/components/Catalog/Chart.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
 import BBChart from "../Chart/BBChart";
-import { CHART_COLORS, topicsList } from "../utils/constants";
+import { CHART_COLORS} from "../utils/constants";
```
Comment (leecalcote) [inline install/docker-extension/ui/src/components/Catalog/Chart.js:2]: Did you fix the low contrast, dark axis displayed on the dark charcoal background?

File: install/docker-extension/ui/src/components/ExtensionComponent/ExtensionComponent.js [hunk 3]
```diff
@@ -482,8 +363,8 @@ const ExtensionsComponent = () => {
         </SectionWrapper>
         {isLoggedIn &&
           (<SectionWrapper>
-            <CatalogChart filter={filter} pattern={pattern} isTheme={isDarkTheme} />
-            <Grid sx={{ backgroundColor: isDarkTheme ? '#666A75' : '#D7DADE', borderRadius: "15px", height: "23rem", display: "flex", justifyContent: "center" }}>
+            <CatalogChart filter={filter} pattern={catalogDesigns} isTheme={isDarkTheme} />
+            <Grid sx={{ backgroundColor: isDarkTheme ? '#666A75' : '#D7DADE', borderRadius: "15px", height: "26rem", display: "flex", justifyContent: "center" }}>
```
Comment (leecalcote) [inline install/docker-extension/ui/src/components/ExtensionComponent/ExtensionComponent.js:367]: Please confirm that this grid is responsive; breaks when it should.

File: install/docker-extension/ui/src/components/utils/constants.js
Comment (leecalcote) [inline install/docker-extension/ui/src/components/utils/constants.js:1]: This change should not have been committed.

File: install/docker-extension/ui/src/index.js
Comment (leecalcote) [inline install/docker-extension/ui/src/index.js:1]: This change should not have been committed.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: install/docker-extension/ui/src/components/Catalog/CatalogCard.js (hunk 1 above)
Comment (Yashsharma1911) [reply install/docker-extension/ui/src/components/Catalog/CatalogCard.js:31]: No we don't need to add name with id, @humblenginr you can remove this instead have 
```
`${mesheryCloudUrl}/catalog/content/${pattern.id}
```

**Comment by humblenginr** [comment]
Closing this in favour of #11599 . Refactors and other features introduced in this PR will be moved over to separate PRs.

**Pull request closed without merging**